Dockerfile

```diff
@@ -3,18 +3,27 @@ FROM debian:bullseye as cppbuild
 RUN apt-get update && apt-get install -y zlib1g-dev libgflags-dev libboost-graph-dev libtsan0 \
       libboost-serialization-dev libboost-test-dev libboost-iostreams-dev libpython3-dev libzmq3-dev \
       libhiredis-dev pybind11-dev wget build-essential git python3 python3-distutils python3-pip file \
-      curl zip unzip tar
+      curl zip unzip tar mono-complete
 RUN wget https://github.com/Kitware/CMake/releases/download/v3.21.3/cmake-3.21.3-Linux-x86_64.sh \
       -q -O /tmp/cmake-install.sh \
       && chmod u+x /tmp/cmake-install.sh \
       && /tmp/cmake-install.sh --skip-license --prefix=/usr/local \
       && rm /tmp/cmake-install.sh
+RUN wget -O /usr/local/bin/nuget.exe https://dist.nuget.org/win-x86-commandline/latest/nuget.exe
 WORKDIR /usr/src/app/
 RUN mkdir ioc-hierarchy-service
 COPY vcpkg/ ioc-hierarchy-service/vcpkg/
-RUN ioc-hierarchy-service/vcpkg/bootstrap-vcpkg.sh
+RUN ioc-hierarchy-service/vcpkg/bootstrap-vcpkg.sh -disableMetrics
+
+
 
 FROM cppbuild as generate_package
+ARG FEED_ACCESSTOKEN
+ARG FEED_URL
+RUN curl -L https://raw.githubusercontent.com/Microsoft/artifacts-credprovider/master/helpers/installcredprovider.sh  | sh
+ENV NUGET_CREDENTIALPROVIDER_SESSIONTOKENCACHE_ENABLED true
+RUN mono /usr/local/bin/nuget.exe sources add -name "ioc-cpp-packages" -Source ${FEED_URL} -Username "docker" -Password ${FEED_ACCESSTOKEN}
+ENV VCPKG_BINARY_SOURCES 'clear;nuget,ioc-cpp-packages,readwrite'
```

Comment:
Does this logic belong in the Dockerfile? This means that the cache cannot be built locally without providing the nuget feed url and access token, doesn't it?

Reply:
From what I understand the feed_url and access token must be passed to the Dockerfile in order to use a private nuget feed as we want here...
Alternatives could be to add (and maintain...) a separate Dockerfile-local
Or add some conditional logic like `RUN if [ "access token and feed url is set ]; then <add feed etc.> ; else continue without feed ; fi`

Others?

Reply:
I do like the solution of fetching from the nuget feed if possible, and continue without it if not. But what will happen in the current implementation if the image is built without providing these arguments?

Reply:
Now builds locally with and without setting `FEED_ACCESSTOKEN`